backend/open_webui/utils/embeddings.py

```
@@ -0,0 +1,90 @@
+
+    # If "direct" flag present, use only that model
+    if getattr(request.state, "direct", False) and hasattr(request.state, "model"):
+        models = {
+            request.state.model["id"]: request.state.model,
+        }
+    else:
+        models = request.app.state.MODELS
+
+    model_id = form_data.get("model")
+    if model_id not in models:
+        raise Exception("Model not found")
```

Ruff (0.11.9)

Create your own exception

(TRY002)